### PR TITLE
feat: upgrade to Next 13

### PR DESCRIPTION
**Pull request by SimenB**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

https://nextjs.org/blog/next-13

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -42,7 +42,7 @@
   },
   "homepage": "https://github.com/fastify/fastify-nextjs#readme",
   "peerDependencies": {
-    "next": "12.x.x"
+    "next": "12.x.x || 13.x.x"
```
Comment (kibertoad) [inline package.json:45]: should we support 12.x if we no longer are running tests on it? wonder if we should just do a semver bump

Comment (mcollina) [reply package.json:45]: I'd also be ok to run tests for both versions in CI. As an example, check out https://github.com/pinojs/hapi-pino/blob/fb705176199cef412de2634e5f554410a79cf852/.github/workflows/test.yml.

Comment (SimenB) [reply package.json:45]: Not sure how to do that with https://github.com/fastify/workflows/blob/main/.github/workflows/plugins-ci.yml?

Comment (mcollina) [reply package.json:45]: let's drop the shared workflow for this

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -2,18 +2,30 @@ name: CI
 
 on:
   push:
-    paths-ignore:
-      - 'docs/**'
-      - '*.md'
+    branches: [ master ]
```
Comment (mcollina) [inline .github/workflows/ci.yml]: please restore the paths ignore and disable the branch check

File: package.json (hunk 1 above)
Comment (SimenB) [reply package.json:45]: @kibertoad @mcollina updated with new job for next 12

File: .github/workflows/ci.yml (hunk 2 above)
Comment (SimenB) [reply .github/workflows/ci.yml]: this is an old diff

**Review by mcollina — APPROVED** [review]
lgtm

**Review by kibertoad — APPROVED** [review]

**Pull request merged**